src/syft/core/node/common/action/get_or_set_property_action.py

```diff
@@ -0,0 +1,193 @@
+        msg_id: Optional[UID] = None,
+    ):
+        super().__init__(address, msg_id=msg_id)
+        self.path = path
+        self.id_at_location = id_at_location
+        self.set_arg = set_arg
+        self._self = _self
+        self.action = action
+        self.args = args
+        self.kwargs = kwargs
+
+    def intersect_keys(
+        self, left: Union[Dict[VerifyKey, UID], None], right: Dict[VerifyKey, UID]
+    ) -> Dict[VerifyKey, UID]:
+        return RunClassMethodAction.intersect_keys(left, right)
+
+    def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
+        method = node.lib_ast(self.path)
+        resolved_self = node.store.get_object(key=self._self.id_at_location)
+        result_read_permissions = resolved_self.read_permissions
+
+        resolved_args = list()
+        for arg in self.args:
+            r_arg = node.store[arg.id_at_location]
+            result_read_permissions = self.intersect_keys(
+                result_read_permissions, r_arg.read_permissions
+            )
+            resolved_args.append(r_arg.data)
+
+        resolved_kwargs = {}
+        for arg_name, arg in self.kwargs.items():
+            r_arg = node.store[arg.id_at_location]
+            result_read_permissions = self.intersect_keys(
+                result_read_permissions, r_arg.read_permissions
+            )
+            resolved_kwargs[arg_name] = r_arg.data
+
+        if not inspect.isdatadescriptor(method):
+            raise ValueError(f"{method} not an actual property!")
+
+        (
+            upcasted_args,
+            upcasted_kwargs,
+        ) = lib.python.util.upcast_args_and_kwargs(resolved_args, resolved_kwargs)
+
+        data = resolved_self.data
+
+        if self.action == PropertyActions.SET:
+            result = method.__set__(data, *upcasted_args, **upcasted_kwargs)
+        elif self.action == PropertyActions.GET:
+            result = method.__get__(data, *upcasted_args, **upcasted_kwargs)
+        elif self.action == PropertyActions.DEL:
+            result = method.__del__(data, *upcasted_args, **upcasted_kwargs)
+        else:
+            raise ValueError(f"{self.action} not a valid action!")
+
+        if lib.python.primitive_factory.isprimitive(value=result):
+            result = lib.python.primitive_factory.PrimitiveFactory.generate_primitive(
+                value=result, id=self.id_at_location
+            )
+        else:
+            if hasattr(result, "id"):
+                try:
+                    if hasattr(result, "_id"):
+                        # set the underlying id
+                        result._id = self.id_at_location
+                    else:
+                        result.id = self.id_at_location
+
+                    assert result.id == self.id_at_location
+                except AttributeError as e:
+                    err = f"Unable to set id on result {type(result)}. {e}"
+                    raise Exception(err)
```

Comment:
Q: Can't we simply raise again the exception (```raise e```)? (or do we want it to be rewritten?)